core/lib/types/src/commitment/mod.rs

```diff
@@ -350,6 +365,23 @@ impl L1BatchAuxiliaryOutput {
                 let state_diffs_hash = H256::from(keccak256(&(state_diffs_packed)));
                 let state_diffs_compressed = compress_state_diffs(state_diffs);
 
+                let blob1_linear_hash = system_logs
+                    .iter()
+                    .find_map(|log| {
+                        (log.0.sender == PUBDATA_CHUNK_PUBLISHER_ADDRESS
+                            && log.0.key == H256::from_low_u64_be(BLOB1_LINEAR_HASH_KEY as u64))
+                        .then_some(log.0.value)
+                    })
+                    .unwrap_or_default();
+                let blob2_linear_hash = system_logs
+                    .iter()
+                    .find_map(|log| {
+                        (log.0.sender == PUBDATA_CHUNK_PUBLISHER_ADDRESS
+                            && log.0.key == H256::from_low_u64_be(BLOB2_LINEAR_HASH_KEY as u64))
+                        .then_some(log.0.value)
+                    })
+                    .unwrap_or_default();
```

Comment:
1. I understand that it's unlikely to become a bottleneck, but still iterating through the logs twice seems to be unnecessary here.
2. Would we want a sanity check that either none or both logs are present?

Probably I'd imagine a simple loop here, like

```rust
let (mut blob1_linear_hash, mut blob2_linear_hash) = (None, None);
for log in &system_logs {
    if (/* first_blob_criterion */) {
        blob1_linear_hash = Some(log.0.value);
    } else if (/* second_blob_criterion */) {
        blob2_linear_hash = Some(log.0.value);
    }
    if blob1_linear_hash.is_some() && blob2_linear_hash.is_some() {
        break;
    }
}

match (&blob1_linear_hash, &blob2_linear_hash) {
   (Some(_), None) | (None, Some(_)) => panic!("Only one blob hash was found in system logs"),
   _ => {}
}
```

Reply:
I'd suggest to keep things as they are... Note that the same `find_map` construction is used multiple times after the changed lines (so to "fully" prevent multiple iterations through the array we'll need 4 mutables variables that we'll track). Also, the total length of the array is unlikely to ever exceed 30 (right now it has 10 elements max)